full-service/src/json_rpc/tx_proposal.rs

```diff
@@ -2,107 +2,96 @@
 
 //! API definition for the TxProposal object.
 
-use crate::json_rpc::unspent_tx_out::UnspentTxOut;
-use mc_mobilecoind_json::data_types::{JsonOutlay, JsonTx, JsonUnspentTxOut};
+use crate::util::b58::{b58_encode_public_address, B58Error};
 
 use serde_derive::{Deserialize, Serialize};
 use std::convert::TryFrom;
 
 #[derive(Deserialize, Serialize, Default, Debug)]
-pub struct TxProposal {
-    pub input_list: Vec<UnspentTxOut>,
-    pub outlay_list: Vec<JsonOutlay>,
-    pub tx: JsonTx,
-    pub fee: String,
-    pub outlay_index_to_tx_out_index: Vec<(String, String)>,
-    pub outlay_confirmation_numbers: Vec<Vec<u8>>,
+pub struct InputTxoJSON {
+    pub tx_out_proto: String,
+    pub value: String,
+    pub token_id: String,
+    pub key_image: String,
 }
 
-impl TryFrom<&mc_mobilecoind::payments::TxProposal> for TxProposal {
-    type Error = String;
-
-    fn try_from(src: &mc_mobilecoind::payments::TxProposal) -> Result<Self, String> {
-        // FIXME: WS-34 - Several unnecessary conversions, but we're leveraging existing
-        // conversion code.
-
-        // First, convert it to the proto
-        let proto_tx_proposal = mc_mobilecoind_api::TxProposal::from(src);
-
-        // Then, convert it to the json representation
-        let json_tx_proposal =
-            mc_mobilecoind_json::data_types::JsonTxProposal::from(&proto_tx_proposal);
+#[derive(Deserialize, Serialize, Default, Debug)]
+pub struct OutputTxoJSON {
+    pub tx_out_proto: String,
+    pub value: String,
+    pub token_id: String,
+    pub recipient_public_address_b58: String,
+    pub confirmation_number: String,
+}
 
-        let outlay_map: Vec<(String, String)> = json_tx_proposal
-            .outlay_index_to_tx_out_index
-            .iter()
-            .map(|(key, val)| (key.to_string(), val.to_string()))
-            .collect();
-        Ok(Self {
-            input_list: json_tx_proposal
-                .input_list
-                .iter()
-                .map(UnspentTxOut::try_from)
-                .collect::<Result<Vec<UnspentTxOut>, String>>()?,
-            outlay_list: json_tx_proposal.outlay_list.clone(),
-            tx: json_tx_proposal.tx.clone(),
-            fee: json_tx_proposal.fee.to_string(),
-            outlay_index_to_tx_out_index: outlay_map,
-            outlay_confirmation_numbers: json_tx_proposal.outlay_confirmation_numbers.clone(),
-        })
-    }
+#[derive(Deserialize, Serialize, Default, Debug)]
+pub struct TxProposalJSON {
```

Comment:
I'm not sure about the rename here with the JSON suffix - is the plan to do that for the json types (which are already namespaced by the json_rpc directory?) 

If we do go this way, can we do it in another PR, as it has a pretty large footprint for changes that are not relevant to multi-token

Reply:
I agree, we should not have the JSON suffix, and simply import it as an alternal name "*JSON" if necessary.